crates/net/eth-wire/src/types/message.rs

```diff
@@ -468,8 +468,22 @@ where
     T: Decodable,
 {
     fn decode(buf: &mut &[u8]) -> alloy_rlp::Result<Self> {
-        let _header = Header::decode(buf)?;
-        Ok(Self { request_id: u64::decode(buf)?, message: T::decode(buf)? })
+        let header = Header::decode(buf)?;
+
+        let initial_length = buf.len();
+        let request_id = u64::decode(buf)?;
+        let message = T::decode(buf)?;
+
+        // Check that the buffer consumed exactly payload_length bytes after decoding the
+        // RequestPair
+        let consumed_len = initial_length - buf.len();
+        if consumed_len != header.payload_length {
+            return Err(alloy_rlp::Error::Custom(
+                "Buffer did not consume exactly payload_length bytes",
+            ));
```

Comment:
Let's use RlpError::UnexpectedLength, we've used it in other places where we check the consumed bytes vs the payload length

Reply:
Done.